### PR TITLE
initial edits for consistency, formatting, and clarity

### PR DESCRIPTION
**Pull request by christinerose**
Signed-off-by: Christine Rose <professor.rose@gmail.com>

Mostly edits in formatting and consistency. Some changes at the beginning of the instructions for clarity. Please feel free to reject changes or ask me anything about why I did a certain thing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
-# eio -- effects based parallel IO for OCaml
+# Eio -- Effects-Based Parallel IO for OCaml
 
 This library implements an effects-based direct-style IO
-stack for multicore OCaml.
+stack for Multicore OCaml. This is a guide for people trying out 
+an experimental branch of OCaml, so it assumes some familiarity with the tools. 
+That said, we'd like to also make it accessible OCaml newcomers. If you're new 
+to OCaml and Opam, please [start here](https://ocaml.org/docs/install.html).
```
[the remark this answers is not in the document]
Comment (talex5) [reply README.md:7]: I guess just take it out, as we already have instructions for installing the effects version below.

**Review by talex5 — COMMENTED** [review]
I've merged the bits that were ready as #79.

I'm not sure whether the remaining bits should just be left as they were, or whether they need rewriting in some other way to be both clear and correct.

File: README.md [hunk 2]
```diff
@@ -267,24 +276,24 @@ Turning off the parent switch will also turn off the child switch, but turning o
...
-or to exit the whole application using the top-level switch.
+or it exits the whole application using the top-level switch.
 
-## Design note: results vs exceptions
+## Design Note: Results vs Exceptions
 
 The OCaml standard library uses exceptions to report errors in most cases.
 Many libraries instead use the `result` type, which has the advantage of tracking the possible errors in the type system.
-However, using `result` is slower, as it requires more allocations, and explicit code to propagate errors.
+However, using `result` is slower because it requires more allocations and explicit code to propagate errors.
```
[the remark this answers is not in the document]
Comment (christinerose) [reply README.md:285]: Definitely keep it as it was if my suggestion makes it confusing!

File: README.md [hunk 3]
```diff
@@ -416,31 +425,31 @@ Imagine this is a large program and we want to know:
...
+and may allow for an Ocap extension to the language in the future.
 See [Emily][] for a previous attempt at this.
 
-## Filesystem access
+## Filesystem Access
 
 Access to the filesystem is also controlled by capabilities, and `env` provides two:
 
-- `fs` provides full access (just like OCaml's stdlib).
+- `fs` provides full access (just like OCaml's `stdlib`).
```
[the remark this answers is not in the document]
Comment (christinerose) [reply README.md:452]: Oh that's good to know!

File: README.md [hunk 4]
```diff
@@ -579,19 +588,19 @@ Notes:
...
   This is why we use `Fibre.both` to create multiple fibres.
 - `run_compute_unsafe` does not start an event loop in the new domain, so it cannot perform IO or create fibres. There will be a separate API for that in the future.
 
-## Design note: thread-safety
+## Design Note: Thread-Safety
 
-OCaml spent the first 25 years of its existence without multicore support, and so most libraries are not thread-safe.
-Even in languages which had parallelism from the beginning thread safety is a very common cause of bugs.
-Eio therefore defaults to a conservative model, in which each mutable value is owned and used by a single domain.
+OCaml spent the first 25 years of its existence without Multicore support, and so most libraries are not thread-safe.
```
[the remark this answers is not in the document]
Comment (christinerose) [reply README.md:601]: Good to know! I've been working with an engineer on Muilticore OCaml, so I got used to capitalising it!

**Comment by christinerose** [comment]
If you want me to have another go at the other parts, I'm happy to do so. Whichever way you think it's more clear.

**Comment by talex5** [comment]
(#79 now merged)

**Pull request closed without merging**